contracts/Holograph.sol

```diff
@@ -107,130 +107,243 @@ import "./abstract/Initializable.sol";
 import "./interface/IInitializable.sol";
 import "./interface/IHolograph.sol";
 
+/**
+ * @title Holograph Protocol
+ * @author https://github.com/holographxyz
+ * @notice This is the primary Holograph Protocol smart contract.
+ * @dev This contract stores a reference to all the primary modules and variables of the protocol.
+ */
 contract Holograph is Admin, Initializable, IHolograph {
-  bytes32 constant _chainTypeSlot = 0x559ee1ab52f1e7dce7933c1945873a89fe5a5eb9f9bac817ed5387cc33fb88b9;
   bytes32 constant _bridgeSlot = 0xeb87cbb21687feb327e3d58c6c16d552231d12c7a0e8115042a4165fac8a77f9;
+  bytes32 constant _chainIdSlot = 0x7651bfc11f7485d07ab2b41c1312e2007c8cb7efb0f7352a6dee4a1153eebab2;
   bytes32 constant _factorySlot = 0xa49f20855ba576e09d13c8041c8039fa655356ea27f6c40f1ec46a4301cd5b23;
+  bytes32 constant _holographChainIdSlot = 0xd840a780c26e07edc6e1ee2eaa6f134ed5488dbd762614116653cee8542a3844;
   bytes32 constant _interfacesSlot = 0xbd3084b8c09da87ad159c247a60e209784196be2530cecbbd8f337fdd1848827;
   bytes32 constant _operatorSlot = 0x7caba557ad34138fa3b7e43fb574e0e6cc10481c3073e0dffbc560db81b5c60f;
   bytes32 constant _registrySlot = 0xce8e75d5c5227ce29a4ee170160bb296e5dea6934b80a9bd723f7ef1e7c850e7;
   bytes32 constant _treasurySlot = 0x4215e7a38d75164ca078bbd61d0992cdeb1ba16f3b3ead5944966d3e4080e8b6;
+  bytes32 constant _utilityTokenSlot = 0xbf76518d46db472b71aa7677a0908b8016f3dee568415ffa24055f9a670f9c37;
 
+  /**
+   * @dev Constructor is left empty and init is used instead.
+   */
   constructor() {}
 
+  /**
+   * @notice Used internally to initialize the contract instead of through a constructor.
+   * @dev This function is called by the deployer/factory when creating a contract.
+   */
   function init(bytes memory data) external override returns (bytes4) {
     require(!_isInitialized(), "HOLOGRAPH: already initialized");
     (
-      uint32 chainType,
+      uint32 holographChainId,
       address bridge,
       address factory,
       address interfaces,
       address operator,
       address registry,
-      address treasury
-    ) = abi.decode(data, (uint32, address, address, address, address, address, address));
+      address treasury,
+      address utilityToken
+    ) = abi.decode(data, (uint32, address, address, address, address, address, address, address));
     assembly {
       sstore(_adminSlot, origin())
-      sstore(_chainTypeSlot, chainType)
       sstore(_bridgeSlot, bridge)
+      sstore(_chainIdSlot, chainid())
       sstore(_factorySlot, factory)
+      sstore(_holographChainIdSlot, holographChainId)
       sstore(_interfacesSlot, interfaces)
       sstore(_operatorSlot, operator)
       sstore(_registrySlot, registry)
       sstore(_treasurySlot, treasury)
+      sstore(_utilityTokenSlot, utilityToken)
     }
     _setInitialized();
     return IInitializable.init.selector;
   }
 
-  function getChainType() external view returns (uint32 chainType) {
+  /**
```

Comment:
Love all the new comments!

Reply:
It's still a work in progress, starting little by little to document all the things.